e2e/setup/setup_test.go

```diff
@@ -0,0 +1,347 @@
+			return fmt.Errorf(
+				"Error while fetching pods; wanted %v Pods but got %v", natsCR.Spec.Cluster.Size, pods.Items,
+			)
+		}
+
+		// Check if all Pods are ready (the status.conditions array has an entry with .type="Ready" and .status="True").
+		for _, pod := range pods.Items {
+			foundReadyCondition := false
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type != "Ready" {
+					continue
+				}
+				foundReadyCondition = true
+				if cond.Status != "True" {
+					return fmt.Errorf(
+						"Pod %s has 'Ready' conditon '%s' but wanted 'True'", pod.GetName(), cond.Status,
+					)
+				}
+			}
+			if !foundReadyCondition {
+				return fmt.Errorf("Could not find 'Ready' condition for Pod %s", pod.GetName())
+			}
+		}
+
+		// Everything is fine.
+		return nil
+	})
+	require.NoError(t, err)
+}
+
+// Test PVCs will test if any PVCs can be found, if their number is equal to the NATS CR's `spec.cluster.size` and if
+// they all have the right size, as defined in `spec.jetStream.fileStorage`.
+func Test_PVCs(t *testing.T) {
+	t.Parallel()
+
+	// Get the PersistentVolumeClaims --PVCs-- and test them.
+	ctx := context.TODO()
+	var pvcs *v1.PersistentVolumeClaimList
+	err := Retry(attempts, interval, func() error {
+		// RetryGet PVCs via a label.
+		var err error
+		pvcs, err = RetryGet(attempts, interval, func() (*v1.PersistentVolumeClaimList, error) {
+			return clientSet.CoreV1().PersistentVolumeClaims(NamespaceName).List(ctx, PVCListOpts())
+		})
+		if err != nil {
+			return err
+		}
+
+		// Check if the amount of PVCs is equal to the spec.cluster.size in the NATS CR. We do this in the retry,
+		// because it may take some time for all PVCs to be there.
+		want, actual := NATSCR().Spec.Cluster.Size, len(pvcs.Items)
+		if want != actual {
+			return fmt.Errorf("error while fetching PVSs; wanted %v PVCs but got %v", want, actual)
+		}
+
+		// Everything is fine.
+		return nil
+	})
+	require.NoError(t, err)
+
+	// Compare the PVC's sizes with the definition in the CRD.
+	for _, pvc := range pvcs.Items {
+		size := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		require.True(t, size.Equal(NATSCR().Spec.FileStorage.Size))
+	}
+}
+
+// Test_Secret tests if the Secret was created.
+func Test_Secret(t *testing.T) {
+	t.Parallel()
+	ctx := context.TODO()
+	err := Retry(attempts, interval, func() error {
+		_, secErr := clientSet.CoreV1().Secrets(NamespaceName).Get(ctx, SecretName, metav1.GetOptions{})
+		if secErr != nil {
+			return secErr
+		}
+		return nil
+	})
+	require.NoError(t, err)
+}
+
+func getNATSCR(ctx context.Context, name, namespace string) (*natsv1alpha1.NATS, error) {
+	var natsCR natsv1alpha1.NATS
+	err := k8sClient.Get(ctx, k8stypes.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, &natsCR)
+	return &natsCR, err
+}
+
+func cmToMap(cm string) map[string]string {
+	lines := strings.Split(cm, "\n")
+
+	cmMap := make(map[string]string)
+	for _, line := range lines {
+		if strings.Contains(line, ": ") {
```

Comment:
(nit) The `if strings.Contains(line, ": ")` is a redundant check, because checking `if len(l) < 2` is enough.

Reply:
removed